libs/api/domains/document-provider/src/lib/document-provider.resolver.ts

```diff
@@ -0,0 +1,37 @@
+import { Args, Mutation, Resolver } from '@nestjs/graphql'
+import { ClientCredentials } from './models/clientCredentials.model'
+import { AudienceAndScope } from './models/audienceAndScope.model'
+import { DocumentProviderService } from './document-provider.service'
+import { TestResult } from './models/testResult.model'
+import { RunEndpointTestsInput } from './dto/runEndpointTests.input'
+import { RegisterEndpointInput } from './dto/registerEndpoint.input'
+import { RegisterProviderInput } from './dto/registerProvider.input'
+
+@Resolver()
+export class DocumentProviderResolver {
+  constructor(private documentProviderService: DocumentProviderService) {}
+
+  @Mutation(() => ClientCredentials)
+  async registerProvider(
+    @Args('input') input: RegisterProviderInput,
+  ): Promise<ClientCredentials> {
+    return this.documentProviderService.registerProvider(input.nationalId)
```

Comment:
you could read the national id from the auth token that associates the request

Reply:
that's true and we will.. just added this for now since there will be some parameters but we don't know which ones atm..